src/python/tests_extended/test_tensor_based.py

```diff
@@ -0,0 +1,590 @@
+                                              window_calculation='Min',
+                                              max_window_size=1,
+                                              horizon=1),
+    'RollingWindow_Simple_Max': RollingWindow(columns={'colA1': 'colA'},
+                                              grain_column=['grainA'], 
+                                              window_calculation='Max',
+                                              max_window_size=1,
+                                              horizon=1),
+    'RollingWindow_Multiple_Grain': RollingWindow(columns={'colA1': 'colA'},
+                                                  grain_column=['grainA'], 
+                                                  window_calculation='Mean',
+                                                  max_window_size=1,
+                                                  horizon=1),
+    'RollingWindow_NonString_Grain': RollingWindow(columns={'colA1': 'colA'},
+                                                   grain_column=['grainA'], 
+                                                   window_calculation='Mean',
+                                                   max_window_size=1,
+                                                   horizon=1),
+    'LagLead_Simple_Lag': LagLeadOperator(columns={'colA1': 'colA'},
+                                          grain_columns=['grainA'], 
+                                          offsets=[-2, -1],
+                                          horizon=2),
+    'LagLead_Simple_Lead': LagLeadOperator(columns={'colA1': 'colA'},
+                                           grain_columns=['grainA'], 
+                                           offsets=[1, 2],
+                                           horizon=2),
+    'LagLead_Complex': LagLeadOperator(columns={'colA1': 'colA'},
+                                       grain_columns=['grainA'], 
+                                       offsets=[-2, -1, 1, 2],
+                                       horizon=2),
+    'ShortGrainDropper': ShortDrop(columns={'colA1': 'colA'},
+                                   grain_columns=['grainA'],
+                                   min_rows=2),
+    'RollingWin_Pivot_Integration': Pipeline([
+        RollingWindow(columns={'colA1': 'colA'},
+                           grain_column=['grainA'], 
+                           window_calculation='Mean',
+                           max_window_size=2,
+                           horizon=2),
+        ForecastingPivot(columns_to_pivot=['colA1'])
+    ]),
+    'Laglead_Pivot_Integration': Pipeline([
+        LagLeadOperator(columns={'colA1': 'colA'},
+                        grain_columns=['grainA'], 
+                        offsets=[-2, -1],
+                        horizon=2),
+        ForecastingPivot(columns_to_pivot=['colA1'])
+    ]),
+}
+TRAINING_DATASETS = {
+    'DateTimeSplitter_Simple': pd.DataFrame(data=dict(
```

Comment:
> 'DateTimeSplitter_Simple': pd.DataFrame(data=dict( [](start = 3, length = 51)

Pls change the test pattern as we agreed where input and output data are very close to each other, so one can verify it